docs/tutorial/source-plugin-tutorial.md

```diff
@@ -0,0 +1,977 @@
+
+## How to create a source plugin
+
+### Overview
+
+The plugin in this tutorial will source blog posts and authors from the demo API, link the posts and authors, and take image URLs from the posts and optimize them automatically. You'll be able to configure your plugin in your site's `gatsby-config.js` file and write GraphQL queries to access your plugin's data.
+
+This tutorial builds off of an existing Gatsby site and some data. If you want to follow along with this tutorial, you can find the codebase inside [the examples folder of the Gatsby repository](https://github.com/gatsbyjs/gatsby/tree/master/examples/creating-source-plugins). Once you clone this code, make sure to delete the `source-plugin` and `example-site` folders. Otherwise, the tutorial steps will already be completed.
+
+#### An example API request
+
+To see the API in action, you can run it locally by navigating into the `api` folder, installing dependencies with `npm install`, and starting the server with `npm start`. You will then be able to navigate to a GraphQL playground running at `http://localhost:4000`. This is a GraphQL server running in Node.js and is _separate from Gatsby_, this server could be replaced with a different backend or data source and the patterns in this tutorial would remain the same. Other possible examples could be a REST API, local files, or even a database, so long as you can access data it can be sourced.
+
+If you paste the following query into the left side of the window and press the play button, you should see data for posts with their IDs and descriptions returned:
+
+```graphql
+query {
+  posts {
+    id
+    description
+  }
+}
+```
+
+This data is an example of the data you will source with your plugin.
+
+_You can also see a running version of the GraphQL playground associated with a distinct API at [https://gatsby-source-plugin-api.glitch.me/](https://gatsby-source-plugin-api.glitch.me/), which is running the `api` folder in a Glitch project, like you would when you run `npm start` on your own computer._
+
+#### Plugin behavior
+
+Your plugin will have the following behavior:
+
+- Make an API request to the demo API.
+- Convert the data in the API response to Gatsby's node system.
+- Link the nodes together so you can query for an author on each post.
+- Accept plugin options to customize how your plugin works.
+- Optimize images from Unsplash URLs so they can be used with `gatsby-image`.
+
+### Set up projects for plugin development
+
+You'll need to set up an example site and create a plugin inside it to begin building.
+
+#### Set up an example site
+
+Create a new Gatsby site with the `gatsby new` command, based on the hello world starter.
```

Comment:
i suggest here to add: "leave the api server running and open a new terminal (tab) to create a new Gatsby site."

and " move to the root folder of the example project" (because we was before in the api folder)